store/src/lib.rs

```diff
@@ -216,12 +218,57 @@ pub trait Store<OT: StoreOpenMode, K, V, MKT: MetadataLocalKey>: Send + Sync {
 #[cfg(feature = "directory")]
 mod directory;
 
+pub fn format_xattr(key: &str) -> String {
+    format!("user.{}", key)
+}
+
 #[derive(Debug, Serialize, Deserialize)]
 pub enum StoreConfig {
     #[cfg(feature = "directory")]
     Directory { path: std::path::PathBuf },
 }
 
+/// Storage modes for the Directory, xattrs or metadata files
+pub enum DirectoryStorageMode {
+    Xattr,
+    MetadataFile,
+}
+
+/// Format of FDOmetadata
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FdoMetadata {
+    pub map: HashMap<String, Vec<u8>>,
+}
+
+/// extension used in our metadata files
+pub const FDO_METADATA_EX: &str = "fdo-md";
```

Comment:
just drop `fdo-` as this file is gonna live within an fdo context anyway (not even meant to be passed around)